scripts/lib/releaseNotes.ts

```diff
@@ -150,3 +149,167 @@ function addNewReleaseNoteToc(releaseNotesNode: any, newVersion: string) {
     });
   }
 }
+
+/**
+ * Sorts the release notes dictionary by the patch version
+ * in descending order, e.g. 0.25.2 then 0.25.1.
+ *
+ * Returns a dictionary with the corresponding markdown for
+ * each patch with the entries sorted by release.
+ */
+export function sortReleaseNotesVersions(markdownByPatchVersion: {
+  [id: string]: string;
+}): { [id: string]: string } {
+  // Sorts the entries of markdownByPathVersion by patch in descending order,
+  // returning an array with elements corresponding to the dictionary entries
+  // as arrays of two elements (key and value).
+  // e.g.
+  // makrdownByPathVersion = {"0.45.0": "test 1", "0.46.0": "test 2"}
+  // =>
+  // markdownByPatchSorted = [["0.46.0", "test 2"],["0.45.0", "test 1"]]
+  const markdownByPatchSorted = Object.entries(markdownByPatchVersion).sort(
+    ([version1], [version2]) => {
+      const versionPatch1 = version1.split("rc").slice(0, 1)[0];
+      const versionPatch2 = version2.split("rc").slice(0, 1)[0];
+
+      if (versionPatch1 == versionPatch2) {
+        // The release candidates within the same patch should appear last.
+        // e.g. version 0.45.1rc1 should appear after version 0.45.1.
+        if (version1.length < version2.length) {
+          return -1;
+        } else if (version1.length > version2.length) {
+          return 1;
+        }
+      }
+
+      return version2.localeCompare(version1);
+    },
+  );
+
+  return Object.fromEntries(markdownByPatchSorted);
+}
+
+/**
+ * Process the markdown dividing it into small sections for each patch version
+ *
+ * Returns a tuple:
+ *   1. A Set of strings representing all the minor versions we found in the
+ *      markdown.
+ *   2. A dictionary with the corresponding markdown for each patch version
+ *      found.
+ */
+export function extractMarkdownReleaseNotesPatches(
+  markdown: string,
+): [Set<string>, { [id: string]: string }] {
+  const sectionsSplit = markdown.split(/\n## (?=[0-9])/);
+  const sections: string[] = sectionsSplit.slice(1, sectionsSplit.length);
+
+  const minorVersionsFound = new Set<string>();
+  const markdownByPatchVersion: { [id: string]: string } = {};
+
+  for (let section of sections) {
+    const versionPatch = section.split("\n").slice(0, 1)[0];
+    const versionMinor = versionPatch.split(".").slice(0, 2).join(".");
+
+    // TODO(https://github.com/Qiskit/qiskit/pull/11436). Remove if/when qiskit 0.45
+    // has been moved into a historical folder.
+    if (+versionMinor < 0.45) {
+      continue;
+    }
```

Comment:
This PR has been merged and cherry-picked to the stable/0.45 branch.

Reply:
Removed! Thanks Eric!